### ISSUE TITLE
[Bug]: LT05 false positive reports 949 > 88 on a 74-character line

### ISSUE THREAD

**Issue by Duy-X** [quarylabs/sqruff#3122]
## What happened?

Sqruff 0.39.0 reports an impossible `layout.long_lines` violation on a Jinja-templated Snowflake SQL file.

Minimal reproduction:

`.sqruff`

```ini
[sqruff]
templater = jinja
dialect = snowflake
rules = all
max_line_length = 88
exclude_rules = references.from

[sqruff:templater:jinja]
apply_dbt_builtins = True

[sqruff:rules:layout.long_lines]
ignore_comment_clauses = True
ignore_comment_lines = True
```

`repro.sql`

```sql
with
    totals as (
        select record_id, sum(metric_value) as total_value
        from {{ ref("upstream_metrics") }}
        where endswith(year_code, '1')
        group by record_id
    )

select
    base.*,
    region_lookup.* exclude (region_id),
    concept_lookup.concept_name,
    {% if var("feature_flag_enabled", true) %}
        optional_lookup.optional_attribute as selected_optional_attribute,
    {% endif %}
    totals.total_value
from base
inner join {{ ref("region_lookup") }} as region_lookup on base.region_code = region_lookup.region_id
left join concept_lookup as concept_lookup on base.record_id = concept_lookup.record_id
{% if var("feature_flag_enabled", true) %}
    left join {{ ref("optional_lookup") }} as optional_lookup on base.optional_code = optional_lookup.optional_code
{% endif %}
left join totals as totals on base.record_id = totals.record_id
where base.created_at >= to_timestamp_ntz('2012-08-06')
order by base.record_id
```

Command:

```bash
sqruff lint repro.sql
```

Actual result:

```text
L:  14 | P:  33 | LT05 | Line is too long (949 > 88). [layout.long_lines]
```

Expected result:

- no `LT05` violation on that line, or
- a reported line length that matches the source file

Why this looks incorrect:

```sql
        optional_lookup.optional_attribute as selected_optional_attribute,
```

That source line is 74 characters long, not 949.

This looks like a rendered-to-source mapping bug around Jinja `{% if %}` blocks.

## Version

sqruff 0.39.0

## Checked in Playground

- [ ] I have checked this bug in the playground.
